### ISSUE TITLE
Add "Inline macro" code action

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode-cpptools#4183]
@sean-mcmanus Could you please add a context-menu option (or even further - a shortcut) to expand macro in-place, right inside the code. This is currently available in Clion and I think it is very easy to implement. Just substitute the macro call with the string that is are currently appearing in `Expands to:` tooltip.

For example:
When you hover over `FOO(ss)` and click the RMB you'll see an item "Expand macro".
```c++
#define FOO(bar) foo##bar
int main() {
   FOO(ss);
}
```
And after you click "Expand macro" you get the expanded version of the macro you clicked on in place where it was:
```c++
#define FOO(bar) foo##bar
int main() {
   fooss;
}
```

_Originally posted by @Veetaha in https://github.com/microsoft/vscode-cpptools/issues/1734#issuecomment-527243945_

**Comment by julianxhokaxhiu** [comment]
Is there any progress on this? There is actually no IDE or editor who does such a thing, and it would make it great to have it on VSCode. Please consider implementing this ASAP!

**Comment by sean-mcmanus** [comment]
It's not on our schedule to implement yet. More upvotes would help.

**Comment by xahon** [comment]
@julianxhokaxhiu Clion does. I'm sure VS does also

**Comment by Trass3r** [comment]
You can get it by using clangd.

**Comment by sean-mcmanus** [comment]
This is available with https://github.com/microsoft/vscode-cpptools/releases/tag/1.5.0-insiders2 . To use it, hover over the macro (which generates the hover expansion data), then click on the macro (which triggers the code action), and the click the lightbulb or use Ctrl+. to and select "Inline macro". Let us know if you hit any bugs with it or have suggestions to improve the feature (preferably by filing a separate issue, so we can track it better).
![image](https://user-images.githubusercontent.com/19859882/122844620-e6068580-d2b6-11eb-95e6-cdf5a9110885.png)

**Comment by ZoeGM1997** [comment]
Hi, I tried to use that macro inline feature. When I hovered over the macro and clicked on the macro, the Inline macro button did not show. Is there any setting controls the inline button or I did it in a wrong way? 
`C/C++ version: 1.5.1`
```
Vscode : Version: 1.58.2 (user setup)
Commit: c3f126316369cd610563c75b1b1725e0679adfb3
Date: 2021-07-14T22:10:15.214Z
Electron: 12.0.13
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Windows_NT x64 10.0.19041
```
![inline_macro](https://user-images.githubusercontent.com/47744191/126126408-5e6e8ada-f43c-420c-a080-353021894868.gif)
